<a href="https://colab.research.google.com/github/AriasVales/DiscreteMathProjects/blob/main/AriasValesAaron_3913_ProjectPart4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project Part #3 : CPSMA 3913 (Dr. Lynam)
Discrete Mathematics

# Choices#

# A. Arias Vales 278816 #

My family and I are going on a family trip. My son loves fruit snacks, and we have 15 different flavors in a bag and when he askes, gets to pull 10 pieces every time. 

In [31]:
fruitFlavors = {"apple", "banana", "orange", "mango", "passionfruit", "cherry", "grape", "papaya", "bloodorange", "grapefruit", "tangerine", "kiwi", "blackberry", "raspberry", "lemon"}

He reaches into the bag to pull his alloted snacks out which can be represented by 
$$ 15*15*15*15*15\dots *15 = 15^{10} $$





In [35]:
15**10

576650390625

we have 10 pieces with 5 different flavors
$$
\binom{n+k-1}{k-1} = \binom{n+k-1}{n}
$$

In [33]:
order_matters = math.factorial(15) // (math.factorial(15 - 10))
print("Outcomes with order mattering:", order_matters)


Outcomes with order mattering: 10897286400


That narrowed quickly. so we went from 576,650,390,625 different combinations down to just 10,897,286,400. 

In [46]:
# (a) Outcomes with order mattering
order_matters = math.factorial(15) // (math.factorial(15 - 10))
print("Outcomes with order mattering:", order_matters)

Outcomes with order mattering: 10897286400


In [47]:
# (b) Outcomes with order not mattering
order_not_matters = math.comb(15, 10)
print("Outcomes with order not mattering:", order_not_matters)

Outcomes with order not mattering: 3003


In [49]:
# (c) Outcomes with no repeat, order mattering
no_repeat_order_matters = math.factorial(15) // (math.factorial(15 - 10))
print("Outcomes with no repeat and order mattering:", no_repeat_order_matters)

Outcomes with no repeat and order mattering: 10897286400


In [50]:
# (c) Outcomes with no repeat, order not mattering
no_repeat_order_not_matters = math.comb(15, 10)
print("Outcomes with no repeat and order not mattering:", no_repeat_order_not_matters)


Outcomes with no repeat and order not mattering: 3003


In [52]:
# (d) Coefficient of 5th order of (x-3)^9
coefficient = (-3)**(9 - 5) * math.comb(9, 5)
print("Coefficient of 5th order of (x-3)^9:", coefficient)

Coefficient of 5th order of (x-3)^9: 10206


So looking at this I believe this is all correct but I could be worng.....
This is easier to do in code versus on paper because of the length it would take for calculations.  The python code allows us to use it similiar to a calculator. The different outcomes correspond to different combinations or permutations of the fruit snacks in the bag.

(a) The "order mattering" outcome counts all possible sequences of 10 fruit snacks that can be picked from the bag, considering the order of the fruit snacks in the sequence as important.

(b) The "order not mattering" outcome counts all possible combinations of 10 fruit snacks that can be picked from the bag, without considering the order of the fruit snacks in the combination.

(c) The "no repeat, order mattering" outcome counts all possible sequences of 10 fruit snacks that can be picked from the bag, without repeat, and considering the order of the fruit snacks in the sequence as important.

(c) The "no repeat, order not mattering" outcome counts all possible combinations of 10 fruit snacks that can be picked from the bag, without repeat, and without considering the order of the fruit snacks in the combination.

Therefore, the different combinations and permutations lead to different outcomes, which is why each one results in a different number.